main.go

```diff
@@ -949,16 +948,43 @@ func (tf *transformer) writeSourceFile(basename, obfuscated string, content []by
 // parseFiles parses a list of Go files.
 // It supports relative file paths, such as those found in listedPackage.CompiledGoFiles,
 // as long as dir is set to listedPackage.Dir.
-func parseFiles(dir string, paths []string) ([]*ast.File, error) {
-	var files []*ast.File
+func parseFiles(lpkg *listedPackage, dir string, paths []string) (files []*ast.File, err error) {
+	main := lpkg.Name == "main" && lpkg.ForTest == ""
+
 	for _, path := range paths {
 		if !filepath.IsAbs(path) {
 			path = filepath.Join(dir, path)
 		}
-		file, err := parser.ParseFile(fset, path, nil, parser.SkipObjectResolution|parser.ParseComments)
+
+		var src any
+
+		abiFileName := filepath.Join("internal", "abi", "type.go")
+		if strings.HasSuffix(path, abiFileName) {
```

Comment:
we already have `sharedCache.GoEnv.GOROOT`; so compare the filename against `filepath.Join(GOROOT, "src", "internal", "abi", "type.go")` instead.

Or even better: `lpkg.ImportPath == "internal/abi" && filepath.Base(path) == "type.go"`